src/TSHelper.ts

```diff
@@ -684,6 +684,20 @@ export class TSHelper {
         return firstDeclaration === node;
     }
 
+    public static isStandardLibraryDeclaration(declaration: ts.Declaration, program: ts.Program): boolean {
+        const source = declaration.getSourceFile();
+        if (!source) { return false; }
+        return program.isSourceFileDefaultLibrary(source);
+    }
+
+    public static isStandardLibraryType(type: ts.Type, name: string, program: ts.Program): boolean {
+        const symbol = type.symbol;
+        if (!symbol || symbol.escapedName !== name) { return false; }
+        const declaration = symbol.valueDeclaration;
+        if(!declaration) { return false; }
```

Comment:
I think the best way to deal with *Constructor types would be to default to `true` here. If I override one of those types, there will be a `valueDeclaration` set on the symbol here. So if there's no `valueDeclaration` and the names match, then it's probably safe to assume it IS the lib version.
```ts
class StringConstructor {
	test() {}
}
function test() {
	const String = new StringConstructor();
	print(String.test()); // symbol.valueDeclaration will point to my override
}
```

Reply:
Good idea, that way at least collisions become very unlikely.